evb/src/main_s2i.cc

```diff
@@ -38,19 +38,18 @@ int16_t static g_in16AudioDataBuffer[LEN_STFT_HOP * 2];
 void
 audio_frame_callback(ns_audio_config_t *config, uint16_t bytesCollected) {
     uint32_t *pui32_buffer =
-        (uint32_t *)am_hal_audadc_dma_get_buffer(config->audioSystemHandle);
+        (uint32_t *) am_hal_audadc_dma_get_buffer(config->audioSystemHandle);
 
     if (g_audioRecording) {
-        if (g_audioReady) {
+        if (g_audioReady)
             ns_printf("Warning - audio buffer wasnt consumed in time\n");
-        }
-        // Raw PCM data is 32b (14b/channel) - here we only care about one
+
+        // Raw PCM data is 32b (12b/channel) - here we only care about one
         // channel For ringbuffer mode, this loop may feel extraneous, but it is
         // needed because ringbuffers are treated a blocks, so there is no way
         // to convert 32b->16b
-        for (int i = 0; i < config->numSamples; i++) {
+        for (int i = 0; i < config->numSamples; i++) 
             g_in16AudioDataBuffer[i] = (int16_t)(pui32_buffer[i] & 0x0000FFF0);
```

Comment:
If you are using latest neuralSPOT, cache is enabled which will cause glitch. There is a patch here that will improve audio quality: https://github.com/AmbiqAI/neuralSPOT/blob/6db388a6358ff14e77cb7c1535eab062fb5f2c71/examples/basic_tf_stub/src/basic_audio.h#L67

Reply:
fixed.